crates/runtime/wit/deps/wasmcloud/bus.wit

```diff
@@ -6,9 +6,13 @@ interface lattice {
         constructor(namespace: string, %package: string, %interface: string);
     }
 
-    /// Set an optional link name to use for all interfaces specified. This is advanced
-    /// functionality only available within wasmcloud and, as such, is exposed here as part of the
-    /// wasmcloud:bus package. This is used when you are linking multiple of the same interfaces
+    /// Set a link name to use for all interfaces specified. This is advanced functionality only
+    /// available within wasmcloud and, as such, is exposed here as part of the wasmcloud:bus package.
+    /// This is used when you are linking multiple of the same interfaces
     /// (i.e. a keyvalue implementation for caching and another one for secrets) to a component
     set-link-name: func(name: string, interfaces: list<call-target-interface>);
+
+    /// Set a link name to use for all interfaces specified. This is the same as `set-link-name`, but will
+    /// return an error if a link does not exist at runtime with the specified name for the specified interfaces.
+    set-link-name-checked: func(name: string, interfaces: list<call-target-interface>) -> result<_, string>;
```

Comment:
This wit package has not been published (#3101). If we're already OK with marking as a breaking change, then I would prefer to support only one interface function as `set-link-name` that returns a result. Caller's should always be expected to handle this case.

The component crate is v0 (minor should be bumped).

Reply:
The WIT interface version is specified as `1.0.0` and while I realize that an actual tarball containing it was never published, it seems probable that users of this interface exist outside this repository.
`1.0.1` with `set-link-name-checked` gated by `@since` would not break such, i.e. all components targeting `wasmcloud:bus@1.0.0` will keep working

`2.0.0` with `set-link-name` returning a result is another option, which would not break anyone (and we could provide both versions of the interface)